src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASING_RATE = 1.5;
+    private T[] elements;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList(int initCapacity) {
+        if (initCapacity <= 0) {
+            throw new IllegalArgumentException("IllegalArgument: " + initCapacity);
+        }
+        this.elements = (T[]) new Object[initCapacity];
+    }
 
+    public ArrayList() {
+        this(DEFAULT_CAPACITY);
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T element) {
+        resizeArray();
+        elements[size] = element;
+        size++;
+    }
 
+    @Override
+    public void add(T element, int index) {
+        if (index > elements.length || index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index more than array size");
```

Comment:
lets make excpetion message more informative and add some info about `index` and current size